js/http.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @param {url} url - The url we will be doing executing
+ * @param {responseType} responseType - The response type expected
+ */
+function baseGetRequest(url, responseType) {
+    return new Promise(function(resolve, reject) {
+        var req = new XMLHttpRequest();
+        req.open('GET', url);
+        req.responseType = responseType;
+
+        req.onload = function() {
+            var response = req.responseXML ? req.responseXML : req.response;
+            if (response && response.errorMessages && response.errorMessages.length > 0) {
+                reject(response.errorMessages[0]);
+                return;
+            }
+            resolve(response);
+        };
+
+        req.onerror = function(error) {
```

Comment:
NOTE TO SELF: Ask candidate why we have this error handler here and an error event listener blow (line 30)